File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {
    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 10) return true;
        
        int copy = numberToCheck;
        int digits = 0;
        int temp = numberToCheck;
        while (temp > 0) {
            temp /= 10;
            digits++;
        }
        temp = numberToCheck;
        int sum = 0;
        while(temp > 0) {
            int digit = temp % 10;
            int power = 1;
            for (int i = 0; i < digits; i++) {
                power *= digit;
            }
            sum += power;
            if (sum > copy) return false;
            temp /= 10;
        }
        return sum == copy;
    }
}

File: solutions/python/armstrong-numbers/3/armstrong_numbers.py
def is_armstrong_number(number):
    if number < 10:
        return True

    original = number
    digits = 0
    temp = number

    while temp > 0:
        temp //= 10
        digits += 1

    temp = number
    total = 0
    while temp > 0:
        digit = temp % 10
        power = 1
        for _ in range (digits):
            power *= digit

        total += power
        if total > original:
            return False

        temp //= 10
    return total == original
